anesthetic/_matplotlib/misc.py

```diff
@@ -0,0 +1,16 @@
+import pandas.plotting._matplotlib.misc as misc
+from anesthetic.weighted_pandas import _WeightedObject
+
+
+def scatter_matrix(frame, *args, **kwargs):
+    # noqa: disable=D103
+    if isinstance(frame, _WeightedObject):
+        frame = frame.compress()
```

Comment:
Compute a weighted scatter matrix by compressing to the channel capacity (could technically do slightly better than this by incorporating weights into the KDE/hist diagonal, but this would involve re-writing the whole function, when we have more functionality in `plot_2d`